syntax = "proto3";

package ibbridge;

import "google/protobuf/timestamp.proto";
import "google/protobuf/duration.proto";

service IbBridge {
  rpc GetStockHistoricalData(GetStockHistoricalDataRequest) returns (stream Bar) {}
}

message GetStockHistoricalDataRequest {
  string symbol = 1;
  string exchange = 2;
  string currency = 3;
  google.protobuf.Timestamp endDate = 4;
}

message Bar {
  google.protobuf.Timestamp timestamp = 1;
  google.protobuf.Duration duration = 2;
  double open = 3;
  double high = 4;
  double low = 5;
  double close = 6;
  int64 volume = 7;
  int64 trades = 8;
  double vwap = 9;
}
